### PR TITLE
Upgraded to Scala 2.13 and removed old unsafe APIs

### PR DESCRIPTION
**Pull request by mpilquist**

### CODE REVIEW

File: src/main/scala/scodec/stream/decode/StreamDecoder.scala [hunk 1]
```diff
@@ -37,11 +35,11 @@ trait StreamDecoder[+A] { self =>
...
-   * Decoding a stream of `A` values from the given `BitVector`.
+   * Decode a stream of `A` values from the given `BitVector`.
    * This function does not retain a reference to `bits`, allowing
    * it to be be garbage collected as the returned stream is traversed.
    */
-  final def decode[F[_]](bits: => BitVector)(implicit F: Effect[F]): Stream[F,A] = Stream.suspend {
+  final def decode[F[_]](bits: => BitVector)(implicit F: Sync[F]): Stream[F,A] = Stream.suspend {
     @volatile var cur = bits // am pretty sure this doesn't need to be volatile, but just being safe
     decoder.translate(new (Cursor ~> F) {
       def apply[X](c: Cursor[X]): F[X] = F.suspend {
```
Comment (SystemFw) [inline src/main/scala/scodec/stream/decode/StreamDecoder.scala]: Why is `F.suspend` needed here btw? stack safety?

Comment (mpilquist) [reply src/main/scala/scodec/stream/decode/StreamDecoder.scala]: Actually, I don't know :) I tried removing it and tests hung, but then later I realized that was just due to some generators that generated very large lists. I'll try again.

Comment (mpilquist) [reply src/main/scala/scodec/stream/decode/StreamDecoder.scala]: OK yeah it works fine so I lifted the `Sync` constraint to `ApplicativeError`

Comment (SystemFw) [reply src/main/scala/scodec/stream/decode/StreamDecoder.scala]: Nice, I've seen the `decodePipeByte` too :)

**Pull request merged**